pkg/pillar/cmd/zedkube/applogs.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build kubevirt
+
+package zedkube
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/kubeapi"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// collectAppLogs - collect App logs from pods which covers both containers and virt-launcher pods
+func collectAppLogs(ctx *zedkubeContext) {
+	sub := ctx.subAppInstanceConfig
+	items := sub.GetAll()
+	if len(items) == 0 {
+		return
+	}
+
+	if ctx.config == nil {
+		config, err := kubeapi.GetKubeConfig()
+		if err != nil {
+			return
+		}
+		ctx.config = config
+	}
+	clientset, err := kubernetes.NewForConfig(ctx.config)
+	if err != nil {
+		log.Errorf("collectAppLogs: can't get clientset %v", err)
+		return
+	}
+
+	// "Thu Aug 17 05:39:04 UTC 2023"
+	timestampRegex := regexp.MustCompile(`(\w{3} \w{3} \d{2} \d{2}:\d{2}:\d{2} \w+ \d{4})`)
+	nowStr := time.Now().String()
+
+	var sinceSec int64
+	sinceSec = logcollectInterval
+	for _, item := range items {
+		aiconfig := item.(types.AppInstanceConfig)
+		contName := base.GetAppKubeName(aiconfig.DisplayName, aiconfig.UUIDandVersion.UUID)
+
+		opt := &corev1.PodLogOptions{}
+		if ctx.appLogStarted {
+			opt = &corev1.PodLogOptions{
+				SinceSeconds: &sinceSec,
+			}
+		} else {
+			ctx.appLogStarted = true
+		}
+		req := clientset.CoreV1().Pods(kubeapi.EVEKubeNameSpace).GetLogs(contName, opt)
+		podLogs, err := req.Stream(context.Background())
+		if err != nil {
+			log.Errorf("collectAppLogs: pod %s, log error %v", contName, err)
+			continue
+		}
+		defer podLogs.Close()
+
+		scanner := bufio.NewScanner(podLogs)
+		for scanner.Scan() {
+			logLine := scanner.Text()
+
+			matches := timestampRegex.FindStringSubmatch(logLine)
+			var timeStr string
+			if len(matches) > 0 {
+				timeStr = matches[0]
+				ts := strings.Split(logLine, timeStr)
+				if len(ts) > 1 {
+					logLine = ts[0]
+				}
+			} else {
+				timeStr = nowStr
+			}
+			// Process and print the log line here
+			aiLogger := ctx.appContainerLogger.WithFields(logrus.Fields{
+				"appuuid":       aiconfig.UUIDandVersion.UUID.String(),
+				"containername": contName,
+				"eventtime":     timeStr,
+			})
+			aiLogger.Infof("%s", logLine)
+		}
+		if scanner.Err() != nil {
+			if scanner.Err() == io.EOF {
```

Comment:
```suggestion
		if scanner.Err() == io.EOF {
```

Reply:
Hmm, shouldn't it also log error and break when there is an error? (It currently doesn't do that.)

Reply:
Sure, let me add an error here.